run.py

```diff
@@ -78,20 +91,75 @@ def cleanup():
     for f in filelist:
         os.remove(f)
 
+def container_name(username):
```

Comment:
You also declare this in the helper. Why not only one function re-used across the framework?

Reply:
(you declare it not as a function, but by using `container_name = "jupyter-" + name`.)
My point is: there should be only one place to rename if I decide to change how the user containers are called.